external/storm-hdfs/src/test/java/org/apache/storm/hdfs/spout/TestHdfsSpout.java

```diff
@@ -192,6 +195,9 @@ public void testEmptySimpleText_ACK() throws Exception {
         Path file1 = new Path(source.toString() + "/file_empty.txt");
         createTextFile(file1, 0);
 
+        //Ensure the second file has a later modified timestamp, as the spout should pick the first file first.
```

Comment:
This is me being a little paranoid about test flakiness. As far as I can tell, the spout reads files ordered by modified timestamp, and if the above two lines run in the same millisecond as the below two, we might read the files in the wrong order.

Reply:
For context, here's the bit in the spout responsible for picking the next file to read https://github.com/apache/storm/blob/master/external/storm-hdfs/src/main/java/org/apache/storm/hdfs/spout/HdfsSpout.java#L588